pkg/cmd/issue/view/presentation_issue_test.go

```diff
@@ -0,0 +1,398 @@
+						},
+					},
+					TotalCount: 1,
+				},
+				ProjectCards: api.ProjectCards{
+					Nodes: []*api.ProjectInfo{
+						{
+							Project: api.ProjectV1ProjectName{
+								Name: "ProjectCardName",
+							},
+							Column: api.ProjectV1ProjectColumn{
+								Name: "ProjectCardColumn",
+							},
+						},
+					},
+					TotalCount: 1,
+				},
+				ProjectItems: api.ProjectItems{
+					Nodes: []*api.ProjectV2Item{
+						{
+							ID: "projectItemID",
+							Project: api.ProjectV2ItemProject{
+								ID:    "projectID",
+								Title: "V2 Project Title",
+							},
+							Status: api.ProjectV2ItemStatus{
+								OptionID: "statusID",
+								Name:     "STATUS",
+							},
+						},
+					},
+				},
+				Milestone: &api.Milestone{
+					Title: "MilestoneTitle",
+				},
+				ReactionGroups: api.ReactionGroups{
+					{
+						Content: "THUMBS_UP",
+						Users: api.ReactionGroupUsers{
+							TotalCount: 1,
+						},
+					},
+				},
+				CreatedAt: createdAt,
+			},
+			expect: PresentationIssue{
+				Title:     "Title",
+				Number:    123,
+				CreatedAt: createdAt,
+				Comments: api.Comments{
+					Nodes: []api.Comment{
+						{
+							Author: api.CommentAuthor{
+								Login: "monalisa",
+							},
+							Body:           "comment body 1",
+							ReactionGroups: api.ReactionGroups{},
+						},
+					},
+					TotalCount: 1,
+				},
+				State:          "OPEN",
+				StateReason:    "",
+				Reactions:      "1 \U0001f44d",
+				Author:         "author",
+				AssigneesList:  "assignee",
+				LabelsList:     "bug",
+				ProjectsList:   "ProjectCardName (ProjectCardColumn), V2 Project Title (STATUS)",
+				MilestoneTitle: "MilestoneTitle",
+				Body:           "",
+				URL:            "github.com/OWNER/REPO/issues/123",
+			},
+		},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			presentationIssue, err := MapApiIssueToPresentationIssue(tc.issue, nil)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			// These are here for development purposes
+			assert.Equal(t, presentationIssue.Title, tc.expect.Title)
+			assert.Equal(t, presentationIssue.Number, tc.expect.Number)
+			assert.Equal(t, presentationIssue.CreatedAt, tc.expect.CreatedAt)
+			assert.Equal(t, presentationIssue.Comments, tc.expect.Comments)
+			assert.Equal(t, presentationIssue.State, tc.expect.State)
+			assert.Equal(t, presentationIssue.Reactions, tc.expect.Reactions)
+			assert.Equal(t, presentationIssue.Author, tc.expect.Author)
+			assert.Equal(t, presentationIssue.AssigneesList, tc.expect.AssigneesList)
+			assert.Equal(t, presentationIssue.LabelsList, tc.expect.LabelsList)
+			assert.Equal(t, presentationIssue.ProjectsList, tc.expect.ProjectsList)
+			assert.Equal(t, presentationIssue.MilestoneTitle, tc.expect.MilestoneTitle)
+			assert.Equal(t, presentationIssue.Body, tc.expect.Body)
+			assert.Equal(t, presentationIssue.URL, tc.expect.URL)
+
+			// This is the actual test
+			assert.Equal(t, presentationIssue, tc.expect)
```

Comment:
Not suggestion this change, but I always wonder `assert` vs `require`.  In this case here, we would get maximum information if any of the assertions fail for a given test, potentially letting us see if we have multiple problems, yeah?

Reply:
Oh, thanks for catching this comment. That's right, as that's my understanding of `assert` vs `require` as well. If we used `require`, the test would exit after the first test failure. Say "Title" failed but the rest of the tests were also failing, we wouldn't get the extra context that the issue may be beyond just "Title". I might go fix "Title" in a weird way, as a result, so getting more information here is better.

That said, I don't usually like overloading tests with this many asserts 😅 The comment of "These are here for development purposes" points out that I used these tests to individually tease out the different pieces of this as I refactored. Do you have an opinion on whether or not we should keep them as they are?